manifests/charts/gateways/istio-egress/values.yaml

```diff
@@ -279,6 +279,9 @@ global:
     # Setting this port to a non-zero value enables STS server.
     servicePort: 0
 
+  # whether to use autoscaling/v2 template for HPA settings
+  autoscalingv2API: true
```

Comment:
Is a user expected to configure this? It would be nice to auto-detect based on their Kubernetes version.

See manifests/charts/gateway/templates/deployment.yaml `ip_unprivileged_port_start` for an example doing this

Reply:
it is for internal usage, i.e. we check the existence of deprecated fields usages from user's IOP and set this autoscalingv2API to false, then render the v2beta1 template instead

Reply:
Can we add a comment to the Helm description saying it's for internal usage? Users look to our Helm values as an API